plugins/Hosting/js/sagas/saga.js

```diff
@@ -204,6 +206,33 @@ function *requestDefaultSettingsSaga() {
 	}
 }
 
+// getHostStatus returns a promise that resolves with `true` if the provided
+// netaddress is connectable, or `false` if it is not connectable. The input
+// 'hostAddr' must be an address in the URL format, eg:
+// tcp://123.456.789.0:9982
+const getHostStatus = (hostAddr) => new Promise((resolve) => {
+	const hostUrl = url.parse(hostAddr)
+	const socket = net.connect(hostUrl.port, hostUrl.hostname)
+	socket.setTimeout(15e3) // 15 second host connectability timeout
```

Comment:
oh wait, this doesn't use the RPC metrics. I see now why you say 'connectable'.

If you look at the RPC metrics you will be able to tell whether people are actually connecting to you or not. Just because you are connectible doesn't mean that your most recent host announcement has the right net address, or that there isn't some other misconfiguration that is preventing renters from seeing you. And after we add proof of burn, you might just not have burned enough coins to be grabbing renter's attention.

So you really do want to be looking at the RPC metrics, they are going to give you a more complete picture than simply pinging yourself.